libsolidity/lsp/LanguageServer.cpp

```diff
@@ -147,6 +150,26 @@ Json::Value LanguageServer::toJson(SourceLocation const& _location)
 
 void LanguageServer::changeConfiguration(Json::Value const& _settings)
 {
+	// The settings item: "file-load-strategy" (enum) defaults to "project-directory" if not (or not correctly) set.
+	// It can be overridden during client's handshake or at runtime, as usual.
+	//
+	// If this value is set to "project-root" (default), all .sol files within the project root will be subject to operations.
+	//
+	// Operations include compiler analysis, but also finding all symbolic references or symbolic renaming.
+	//
+	// If this value is set to "directly-opened-and-on-import", then only currently directly opened files and
+	// those files being imported directly or indirectly will be included in operations.
+	if (_settings["file-load-strategy"])
+	{
+		auto const text = _settings["file-load-strategy"].asString();
```

Comment:
It'll be auto-converted to a string, and if impossible (i.e. it's an JSON object rather than a scalar type), then an `Json::LogicError` is thrown, which will be caught in the `run()` function that reports back to the caller with an error message, saying that the JSON input was invalid.

Reply:
Ah, ok then. Sounds good.